pkg/occlient/occlient_test.go

```diff
@@ -1952,6 +1973,227 @@ func TestNewAppS2I(t *testing.T) {
 	}
 }
 
+func TestIsTagInImageStream(t *testing.T) {
+	tests := []struct {
+		name        string
+		imagestream imagev1.ImageStream
+		imageTag    string
+		wantErr     bool
+		want        bool
+	}{
+		{
+			name:        "Case: Valid image and image tag",
+			imagestream: *fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+			imageTag:    "3.5",
+			want:        true,
+		},
+		{
+			name:        "Case: Invalid image tag",
+			imagestream: *fakeImageStream("bar", "testing", []string{"latest"}),
+			imageTag:    "0.1",
+			want:        false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			got := isTagInImageStream(tt.imagestream, tt.imageTag)
+
+			if got != tt.want {
+				t.Errorf("GetImageStream() = %#v, want %#v\n\n", got, tt)
+			}
+		})
+	}
+}
+func fakeImageStreamImage(imageName string, ports []string) *imagev1.ImageStreamImage {
+	exposedPorts := make(map[string]struct{})
+	var s struct{}
+	for _, port := range ports {
+		exposedPorts[port] = s
+	}
+	return &imagev1.ImageStreamImage{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s@@sha256:9579a93ee", imageName),
+		},
+		Image: imagev1.Image{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "@sha256:9579a93ee",
+			},
+			DockerImageMetadata: runtime.RawExtension{
+				Object: &dockerapiv10.DockerImage{
+					ContainerConfig: dockerapiv10.DockerConfig{
+						ExposedPorts: exposedPorts,
+					},
+				},
+			},
+			DockerImageReference: fmt.Sprintf("docker.io/centos/%s-36-centos7@s@sha256:9579a93ee", imageName),
+		},
+	}
+}
+
+func TestGetExposedPorts(t *testing.T) {
+	tests := []struct {
+		name        string
+		imagestream *imagev1.ImageStream
+		imageTag    string
+		wantErr     bool
+		want        []corev1.ContainerPort
+	}{
+		{
+			name:        "Case: Valid image ports",
+			imagestream: fakeImageStream("python", "openshift", []string{"latest", "3.5"}),
+			imageTag:    "3.5",
+			want: []corev1.ContainerPort{
+				{
+					Name:          fmt.Sprintf("%d-%s", 8080, strings.ToLower(string("tcp"))),
+					ContainerPort: 8080,
+					Protocol:      "TCP",
+				},
+			},
+		},
+		/*
```

Comment:
Reason for commenting this out?

Reply:
Aah commented it out while debugging something.
Will un-comment the test